### PR TITLE
Fix transient CSE exit 9

### LINKED ISSUE #3204: Transient CSE exit code 9 on VMSS with k8s 1.10
I'm seeing about 50% of VMSS k8s 1.10 deployments end with:

```
2018/06/06 20:44:57 Command Output: ERROR: Deployment failed. Correlation ID: xxx. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'vmssCSE'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=9\n[stdout]\n\n[stderr]\n\"."
      }
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: This makes the cluster provisioning wait until cloud-init configuration has completed prior to updating the cache. A part of the cloud-init configuration includes adding apt source (templating), and it appears as though there was a race condition causing transient failures. (creds: @trstringer)

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #3204 

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```release-note
```

### CODE REVIEW

File: parts/k8s/kubernetescustomscript.sh [hunk 1]
```diff
@@ -154,8 +154,13 @@ function installEtcd() {
...
 
 function installDeps() {
+    echo `date`,`hostname`, apt-get_update_begin>>/opt/m
+    apt_get_update || exit $ERR_APT_INSTALL_TIMEOUT
+    echo `date`,`hostname`, apt-get_update_end>>/opt/m
+	# make sure walinuxagent doesn't get updated in the middle of running this script
+	retrycmd_if_failure 20 5 30 apt-mark hold walinuxagent || exit $ERR_HOLD_WALINUXAGENT
     # See https://github.com/kubernetes/kubernetes/blob/master/build/debian-hyperkube-base/Dockerfile#L25-L44
-    apt_get_install 20 30 180 apt-transport-https ca-certificates iptables iproute2 socat util-linux mount ebtables ethtool init-system-helpers nfs-common ceph-common conntrack glusterfs-client ipset jq || exit $ERR_APT_INSTALL_TIMEOUT
+    apt_get_install 20 30 300 apt-transport-https ca-certificates iptables iproute2 socat util-linux mount ebtables ethtool init-system-helpers nfs-common ceph-common conntrack glusterfs-client ipset jq || exit $ERR_APT_INSTALL_TIMEOUT
```
Comment (CecileRobertMichon) [inline parts/k8s/kubernetescustomscript.sh]: @jackfrancis how do we feel about increasing the timeout here?

File: parts/k8s/kubernetescustomscript.sh [hunk 2]
```diff
@@ -154,8 +154,13 @@ function installEtcd() {
 }
 
 function installDeps() {
+    echo `date`,`hostname`, apt-get_update_begin>>/opt/m
+    apt_get_update || exit $ERR_APT_INSTALL_TIMEOUT
+    echo `date`,`hostname`, apt-get_update_end>>/opt/m
+	# make sure walinuxagent doesn't get updated in the middle of running this script
```
Comment (jackfrancis) [inline parts/k8s/kubernetescustomscript.sh]: nit: whitespace indent variance

**Comment by jackfrancis** [comment]
This makes the cluster provisioning wait until cloud-init configuration has completed prior to updating the cache. A part of the cloud-init configuration includes adding apt source (templating), and it appears as though there was a race condition causing transient failures (for more information, please see #3204).

**Comment by CecileRobertMichon** [comment]
@jackfrancis added as PR description

**Pull request merged**